code/game/machinery/slotmachine.dm

```diff
@@ -84,46 +94,48 @@
 		icon_screen = "slots_screen"
 	return ..()
 
-/obj/machinery/computer/slot_machine/attackby(obj/item/I, mob/living/user, params)
-	if(istype(I, /obj/item/coin))
-		var/obj/item/coin/C = I
+/obj/machinery/computer/slot_machine/attackby(obj/item/inserted, mob/living/user, params)
```

Comment:
Move all code inside `attackby()` to `item_interact_with()`. To see why refer https://github.com/tgstation/tgstation/pull/81613#discussion_r1498502689